extensions/dialog/internal.m

```diff
@@ -0,0 +1,388 @@
+///  * [style] can be "NSWarningAlertStyle", "NSInformationalAlertStyle" or "NSCriticalAlertStyle". If something other than these string values is given, it will use "NSWarningAlertStyle".
+/*
+ EXAMPLE:
+ testWebview = hs.webview.newBrowser(hs.geometry.rect(250, 250, 250, 250)):show()
+ testCallbackFn = function(result) print("Callback Result: " .. result) end
+ hs.dialog.webviewAlert(testWebview, testCallbackFn, "Message", "Informative Text", "Button One", "Button Two", "NSCriticalAlertStyle")
+ */
+static int webviewAlert(lua_State *L) {
+    
+    NSString* defaultButton = @"OK";
+    
+    LuaSkin *skin = [LuaSkin shared];
+    [skin checkArgs:LS_TUSERDATA, "hs.webview", LS_TFUNCTION, LS_TSTRING, LS_TSTRING, LS_TSTRING | LS_TOPTIONAL, LS_TSTRING | LS_TOPTIONAL, LS_TSTRING | LS_TOPTIONAL, LS_TBREAK];
+    
+    NSWindow *webview = [skin toNSObjectAtIndex:1];
+    //NSObject *callback = [skin toNSObjectAtIndex:2];
+    NSString *message = [skin toNSObjectAtIndex:3];
+    NSString *informativeText = [skin toNSObjectAtIndex:4];
+    NSString *buttonOne = [skin toNSObjectAtIndex:5];
+    NSString *buttonTwo = [skin toNSObjectAtIndex:6];
+    NSString *style = [skin toNSObjectAtIndex:7];
+    
+    NSAlert *alert = [[NSAlert alloc] init];
+    [alert setMessageText:message];
+    [alert setInformativeText:informativeText];
+    
+    if( buttonOne == nil ){
+        [alert addButtonWithTitle:defaultButton];
+    }
+    else
+    {
+        [alert addButtonWithTitle:buttonOne];
+    }
+    
+    if (buttonTwo != nil ) {
+        [alert addButtonWithTitle:buttonTwo];
+    }
+    
+    if (style == nil){
+        [alert setAlertStyle:NSWarningAlertStyle];
+    }
+    else
+    {
+        if ([style isEqualToString:@"NSWarningAlertStyle"]) {
+            [alert setAlertStyle:NSWarningAlertStyle];
+        }
+        else if ([style isEqualToString:@"NSInformationalAlertStyle"]) {
+            [alert setAlertStyle:NSInformationalAlertStyle];
+        }
+        else if ([style isEqualToString:@"NSCriticalAlertStyle"]) {
+            [alert setAlertStyle:NSCriticalAlertStyle];
+        }
+        else
+        {
+            [alert setAlertStyle:NSWarningAlertStyle];
+        }
+    }
+    
+    [alert beginSheetModalForWindow:webview completionHandler:^(NSModalResponse result){
+        
+        NSString *button = defaultButton;
+        
+        if (result == NSAlertFirstButtonReturn) {
+            if (buttonOne != nil) {
+                button = buttonOne;
+            }
+        }
+        else if (result == NSAlertSecondButtonReturn) {
+            button = buttonTwo;
+        }
+        else
+        {
+            [LuaSkin logError:@"hs.dialog.webviewAlert() - Failed to detect which button was pressed."];
+            lua_pushnil(L) ;
+        }
+        
+        lua_pushvalue(L, 2) ; // push the function passed in as an argument to the top of the stack
```

Comment:
remove this lua_pushvalue line and replace it with:

~~~objc
    [skin pushLuaRef:refTable ref:callbackRef] ; // put the saved function back on the stack
    [skin luaUnref:refTable ref:callbackRef] ; // remove the stored function from the registry; normally we'd do this in a userdata's __gc, but since we're not using a userdata for this, we need to do so within the NSAlert callback context... could go at the end, but since it won't affect the stack at all, I'd do it here so we don't forget
~~~